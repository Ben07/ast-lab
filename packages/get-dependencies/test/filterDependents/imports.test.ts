// @ts-ignore
import { enumerateArgsTestFunction, configArgs } from 'lazy-jest';
import { filterDependents } from './utils';

enumerateArgsTestFunction(
  filterDependents.bind(null, 'imports'),
  configArgs()
  .arg('sources', [
    ['base.js'],
    ['base.js', 'importAlias.js'],
    ['base.js', 'importAll.js'],
    ['base.js', 'importDefault.js'],
    ['base.js', 'importNamed.js'],
    ['base.js', 'noImports.js'],
    ['base.js', 'importCss.js']
  ])
  .arg('targets', [
    { './base': ['a', 'default'] }
  ]),
  'imports'
);