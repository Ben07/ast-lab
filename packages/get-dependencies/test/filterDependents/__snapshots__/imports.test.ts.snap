// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports - (["sources", "targets"]) - [["base.js","importAlias.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
  "/imports/importAlias.js",
]
`;

exports[`imports - (["sources", "targets"]) - [["base.js","importAll.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
  "/imports/importAll.js",
]
`;

exports[`imports - (["sources", "targets"]) - [["base.js","importCss.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
]
`;

exports[`imports - (["sources", "targets"]) - [["base.js","importDefault.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
  "/imports/importDefault.js",
]
`;

exports[`imports - (["sources", "targets"]) - [["base.js","importNamed.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
  "/imports/importNamed.js",
]
`;

exports[`imports - (["sources", "targets"]) - [["base.js","noImports.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
]
`;

exports[`imports - (["sources", "targets"]) - [["base.js"],{"./base":["a","default"]}] 1`] = `
Array [
  "/imports/base.js",
]
`;
